### DISCUSSION TITLE
Device Power Consumption

### DISCUSSION THREAD

**Discussion by 34by151** [jmcollin78/solar_optimizer#57] [Q&A]
Been changing over from pv_excess
So far I really like solar_optimizer

Is there a way to set the device power consumption?

In my Devices I have power_max set to the max consumption
It is working as expected 
It wont turn on the load unless I solar export > power_max 

What I want do is trun the load on when solar export > power consumption

I was tring withot sucess to set power_max to a sensor variable
I think this would do what I want but cant get the "power_max" to accept a sensor value

I thought of changing to a "variable power equipment"  but this apears to control the power output once device is activated
What Im looking to do is a "variable power" activation and then "variable power" deactivation

**Comment by jmcollin78** [comment]
Hello @34by151 ,

I understand your point but I wonder how it can works:  if the device is stopped, by definition its consumed power is 0. So SO will always restart it, then the power will increase, and stop it at next cycle, ...etc

**Reply by 34by151** [reply]
Well yes and no

For some load like the pool and hot water this is the case
For others loads switching them to another source such as a different phase or to off peak power

If the laod is switced to Off Peak (because I need it to keep running)
I still have a consumed power number

Lets say the power_max = 4000w and its on
Then the somthing happens that causes the load to turn off 
I use this to switch it to off peak and keep it running
Optimiser has it swithed off and assuming 4000w is needed to turn back on
Then the load drops to 2000w (running of off peak)
It wont turn back on as it calculating for 4000W 
If it does turn on (solar goes up or other loads go down), I switch it back to the solar phase 

So the pool may be consuming power but not switched to the pahse with the solar

Not sure if this is feasable or not
Can power_max be made to accept a sensor variable?

If it can be made to do that my issue is solved
Id use an average of the load (over a few minutes) as the sensor
Optimiser should use the latest value in the calculation on the next cycle

Hope that not too confusing
And thanks for the grat work, much apriciated

**Reply by jmcollin78** [reply]
> Can power_max be made to accept a sensor variable?

Yes this is feasable but I want to keep my integration as much simplier than possible and avoid implementing too specific features. Let me think about that.

**Reply by 34by151** [reply]
thank you
